r/tools/linuxlibs.R

```diff
@@ -379,40 +379,41 @@ cmake_version <- function(cmd = "cmake") {
   )
 }
 
-with_s3_support <- function(env_vars) {
+with_s3_support <- function(env_vars, cmake) {
   arrow_s3 <- toupper(Sys.getenv("ARROW_S3")) == "ON" || tolower(Sys.getenv("LIBARROW_MINIMAL")) == "false"
   if (arrow_s3) {
     # User wants S3 support. Let's make sure they're not on gcc < 4.9
     # and make sure that we have curl and openssl system libs
-    info <- system(paste(env_vars, "&& $CMAKE --system-information"), intern = TRUE)
+    info <- system(paste(env_vars, cmake, "--system-information"), intern = TRUE)
```

Comment:
`/path/to/cmake` is passed in as `CMAKE` in the env_vars, hence why cmake was invoked as `$CMAKE`. Would `paste("export", env_vars, "&& $CMAKE --system-information")` also solve this?

Reply:
Yeah, that _should_ work though I need to double check.